angular/bootstrap/src/components/progressbar/progressbar.component.ts

```diff
@@ -159,19 +154,22 @@ export class ProgressbarComponent extends BaseWidgetDirective<ProgressbarWidget>
 	 */
 	@Input('auType') type: BSContextualClass | undefined;
 
-	readonly _widget = callWidgetFactory({
-		factory: createProgressbar,
-		widgetName: 'progressbar',
-		defaultConfig: this.defaultSlots,
-		afterInit: () => {
-			useDirectiveForHost(this._widget.directives.ariaDirective);
-		},
-	});
-
-	ngAfterContentChecked(): void {
-		this._widget.patchSlots({
-			children: undefined,
-			structure: this.slotStructureFromContent?.templateRef,
-		});
+	@ViewChild('content', {static: true})
+	slotChildren?: TemplateRef<void>;
+
+	constructor() {
+		super(
+			callWidgetFactory({
+				factory: createProgressbar,
+				widgetName: 'progressbar',
+				defaultConfig,
+				afterInit: (widget) => useDirectiveForHost(widget.directives.ariaDirective),
+				slotTemplates: () => ({
+					children: undefined,
```

Comment:
as discussed, we should either filter out children, or (better) remove the possibility to have both a slot and a ng-content

Reply:
as discussed, it's better to provide a directive for type checking ng-templates